tests/cross/src/test/scala/tests/pc/CompletionScaladocSuite.scala

```diff
@@ -0,0 +1,194 @@
+      |}""".stripMargin,
+    ""
+  )
+
+  checkEdit(
+    "scaladoc-methoddef-edit",
+    """|object A {
+       |  /**@@
+       |  def test1(param1: Int, param2: Int): Int = ???
+       |  def test2(param1: Int, param2: Int, param3: Int): Int = ???
+       |}
+       |""".stripMargin,
+    """|object A {
+       |  /**
+       |  *
+       |  * @param param1 $0
+       |  * @param param2
+       |  * @return
+       |  */
+       |  def test1(param1: Int, param2: Int): Int = ???
+       |  def test2(param1: Int, param2: Int, param3: Int): Int = ???
+       |}
+       |""".stripMargin,
+    filter = _.contains("/** */")
+  )
+
+  checkEdit(
+    "scaladoc-classdef-edit",
+    """|object A {
+       |  /**@@
+       |  class Test1(param1: Int, param2: Int) {}
+       |  class Test2(param1: Int, param2: Int, param3: Int) {}
+       |}
+       |""".stripMargin,
+    """|object A {
+       |  /**
+       |  *
+       |  * @param param1 $0
+       |  * @param param2
+       |  */
+       |  class Test1(param1: Int, param2: Int) {}
+       |  class Test2(param1: Int, param2: Int, param3: Int) {}
+       |}
+       |""".stripMargin,
+    filter = _.contains("/** */")
+  )
+
+  checkEdit(
+    "scaladoc-valdef-edit",
+    """|object A {
+       |  /**@@
+       |  val x = 1
+       |}
+       |""".stripMargin,
+    """|object A {
+       |  /**
+       |  *
+       |  */
+       |  val x = 1
+       |}
+       |""".stripMargin,
+    filter = _.contains("/** */")
+  )
+
+  checkEdit(
+    "scaladoc-valdef-edit",
+    """|object A {
+       |  /**@@
+       |  val x = 1
+       |}
+       |""".stripMargin,
+    """|object A {
+       |  /**
+       |  *
+       |  */
+       |  val x = 1
+       |}
+       |""".stripMargin,
+    filter = _.contains("/** */")
+  )
+
+  checkEdit(
+    "scaladoc-objectdef-edit",
+    """|/**@@
+       |object A {
+       |  // do not calculate scaladoc based on the method
+       |  def test(x: Int): Int = ???
+       |}
+       |""".stripMargin,
+    """|/**
+       |  *
+       |  */
+       |object A {
+       |  // do not calculate scaladoc based on the method
+       |  def test(x: Int): Int = ???
+       |}
+       |""".stripMargin,
+    filter = _.contains("/** */")
+  )
+
+  checkEdit(
+    "scaladoc-objectdef-edit",
+    """|/**@@
+       |object A {
+       |  // do not calculate scaladoc based on the method
+       |  def test(x: Int): Int = ???
+       |}
+       |""".stripMargin,
+    """|/**
+       |  *
+       |  */
+       |object A {
+       |  // do not calculate scaladoc based on the method
+       |  def test(x: Int): Int = ???
+       |}
+       |""".stripMargin,
+    filter = _.contains("/** */")
+  )
+
+  checkEdit(
+    "scaladoc-defdef-nested-edit",
+    """|object A {
+       |  def test(x: Int): Int = {
+       |    /**@@
+       |    def nest(y: Int) = ???
+       |  }
+       |}
+       |""".stripMargin,
+    """|object A {
+       |  def test(x: Int): Int = {
+       |    /**
+       |  *
```

Comment:
indentation is off? We probably need something like `inferIndent` that is used for auto imports completions

Reply:
This actually works fine in VSCode, and I think I know what's going on here. 

VSCode and other LSP clients automatically add indentation when returning a multi-line `textEdit`. Other clients don't.

In a similar case, we add indentation based on `metalsConfig.snippetAutoIndent`, see here for an example:
 https://github.com/scalameta/metals/blob/25641a7ca25593a45243ade7eae084ec9e46c2a4/mtags/src/main/scala/scala/meta/internal/pc/Completions.scala#L1300-L1306